src/plugins/data/public/search/search_service.ts

```diff
@@ -92,22 +95,28 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
 
   public setup(
     core: CoreSetup,
-    { packageInfo, query, getInternalStartServices }: SearchServiceSetupDependencies
+    { expressions, packageInfo, query, getInternalStartServices }: SearchServiceSetupDependencies
   ): ISearchSetup {
     this.esClient = getEsClient(core.injectedMetadata, core.http, packageInfo);
     this.registerSearchStrategyProvider(SYNC_SEARCH_STRATEGY, syncSearchStrategyProvider);
     this.registerSearchStrategyProvider(ES_SEARCH_STRATEGY, esSearchStrategyProvider);
 
     const aggTypesSetup = this.aggTypesRegistry.setup();
+
+    // register each agg type
     const aggTypes = getAggTypes({
       query,
       uiSettings: core.uiSettings,
       getInternalStartServices,
     });
-
     aggTypes.buckets.forEach(b => aggTypesSetup.registerBucket(b));
     aggTypes.metrics.forEach(m => aggTypesSetup.registerMetric(m));
 
+    // register expression functions for each agg type
+    const aggFunctions = getAggTypesFunctions();
+    aggFunctions.buckets.forEach(b => expressions.registerFunction(b));
```

Comment:
why are we splitting the functions into 'buckets' and 'metrics' ? seems irrelevant at this point.

Reply:
only did that for consistency because it is how the aggTypes are registered above, but i will go back and simplify this. agreed that it is isn't really relevant